Documentation: clarify label matching (#55650)
grafana/grafana: docs/sources/alerting/fundamentals/annotation-label/labels-and-label-matchers.md

| `=~`     | Select labels that regex-match the value.          |
| `!~`     | Select labels that do not regex-match the value.   |

In case of multiple label matchers, they are combined using AND logical operator, meaning all matchers need to match in order to link a rule to a policy.

> ## Example of a label matcher
> 
> Imagine we've defined the following set of labels for our alert.

Reviewer: ```suggestion
## Example scenario

If you define the following set of labels for your alert:
```

- A label matcher defined as `foo!=bar` will _not_ match this alert rule.
- A label matcher defined as `id=~[0-9]+` will match this alert rule.
- A label matcher defined as `baz!~[0-9]+` will match this alert rule.

> - Two label matchers defined as `foo=bar` and `id=~[0-9]+` will match this alert rule.

Reviewer 2: I wonder if this can be written more clearly.

Reviewer: I would also delete "In this situation...".

Author: Changed.

> - A label matcher defined as `foo!=bar` will _not_ match this alert rule.
> - A label matcher defined as `id=~[0-9]+` will match this alert rule.
> - A label matcher defined as `baz!~[0-9]+` will match this alert rule.
> - Two label matchers defined as `foo=bar` and `id=~[0-9]+` will match this alert rule.

Reviewer: ```suggestion
- A label matcher defined as `foo!=bar` matches this alert rule.
- A label matcher defined as `id=~[0-9]+` matches this alert rule.
- A label matcher defined as `baz!~[0-9]+` matches this alert rule.
- Two label matchers defined as `foo=bar` and `id=~[0-9]+` match this alert rule.
```

Author: In this suggestion I think you accidentally removed negation in `foo!=bar` case, so I did not accept it, but I did a similar modification.
